core/autocrypt/pgpycrypto.py

```diff
@@ -0,0 +1,378 @@
+        """Init PGPyCrypto class.
+
+        :param homedir: home dir
+        :type key: str
+        :param gpgpath: kept for compatibility with BinGPG API
+        :type gpgpath: str
+
+        """
+        # NOTE: called as .._pgpy.. to know that is instance of PGPKey
+        self.pgpydir = homedir
+        self.memkr = PGPKeyring()
+        self._ensure_init()
+
+    def __str__(self):
+        return "PGPyCrypto(homedir={homedir!r})".format(
+            homedir=self.pgpydir)
+
+    def _ensure_init(self):
+        if self.pgpydir is None:
+            return
+        if not os.path.exists(self.pgpydir):
+            os.mkdir(self.pgpydir)
+            os.chmod(self.pgpydir, 0o700)
+        self._loadkr()
+
+    def _loadkr(self):
+        keyfiles = glob.glob(os.path.join(self.pgpydir, '*.asc'))
+        self.memkr.load(keyfiles)
+
+    def _savekr(self):
+        # NOTE: saving secret keys in clear in the filesystem is a secuirty
+        # risk, but the generated keys with GPG are not passphrase protected
+        # and PGPY does not implement yet filesystem key ring
+        for fp in self.memkr.fingerprints():
+            with self.memkr.key(fp) as key:
+                self._save_key_to_file(key)
+
+    def _key_path(self, key):
+        ext = ''
+        if key.is_public is False:
+            ext = '.sec'
+        ext += '.asc'
+        keypath = os.path.join(self.pgpydir, key.fingerprint.keyid + ext)
+        return keypath
+
+    def _load_key_into_kr(self, key):
+        keypath = self._save_key_to_file(key)
+        self.memkr.load(keypath)
+
+    def _save_key_to_file(self, key):
+        keypath = self._key_path(key)
+        with open(keypath, 'wb') as fd:
+            fd.write(key_bytes(key))
+        return keypath
+
+    def _gen_skey_usage_all(self, emailadr):
+        skey = PGPKey.new(SKEY_ALG, KEY_SIZE)
+        # NOTE: pgpy implements separate attributes for name and e-mail
+        # address. Name is mandatory.
+        # Here e-mail address is used for the attribute name .
+        # If name attribute would be set to empty string
+        # and email to the e-mail address, the uid would be
+        # ' <e-mail address>', for instance:
+        # " <alice@testsuite.autocrypt.org>" - which we do not want.
+        uid = PGPUID.new(emailadr)
+        skey.add_uid(uid, usage=SKEY_USAGE_ALL, **SKEY_ARGS)
+        return skey
+
+    def _gen_ssubkey(self):
+        # NOTE: the uid for the subkeys can be obtained with .parent,
+        # but, unlike keys generated with gpg, it's not printed when imported
+        # in gpg keyring and run --fingerprint
+        # in case of adding uid to the subkey, it raises are currently some
+        # exceptions depending on which are the arguments used, which are not
+        # clear from the documentation.
+        ssubkey = PGPKey.new(SKEY_ALG, KEY_SIZE)
+        return ssubkey
+
+    def _gen_skey_with_subkey(self, emailadr='alice@testsuite.autocrypt.org'):
```

Comment:
Another dummy email address, also in next function.

@juga0 if you can quickly fix these would be great. Otherwise i am happy to merge and fix afterwards.